### PR TITLE
feat(stdlib): Add `addCharFromCodePoint` to Buffer module

### PR DESCRIPTION
**Pull request by spotandjake**
This add `Buffer.addCharFromCodePoint` to the `Buffer` module, this is used in #1133. This is very similar to doing:
```
Buffer.addUint32(Char.code('a'))
```
but it handles the correct byte count of chars.

Internally I chose to just call `addChar` this leaves less surface area for bugs to crop up, and given char is a stack type we are not giving up a ton of perf. Well once #1879 is completed.

### CODE REVIEW

File: stdlib/buffer.gr [hunk 1]
```diff
@@ -322,6 +322,18 @@ provide let addChar = (char, buffer) => {
   addString(Char.toString(char), buffer)
 }
 
+/**
+ * Appends a char from a code point to a buffer.
+ *
+ * @param codePoint: The codePoint to append to the buffer
```
Comment (ospencer) [inline stdlib/buffer.gr:328]:
```suggestion
 * @param codePoint: The code point to append to the buffer
```

Comment (spotandjake) [reply stdlib/buffer.gr]: Done

**Review by jozanza — APPROVED** [review]

**Review by peblair — APPROVED** [review]

File: stdlib/buffer.gr [hunk 2]
```diff
@@ -322,6 +322,18 @@ provide let addChar = (char, buffer) => {
   addString(Char.toString(char), buffer)
 }
 
+/**
+ * Appends a char from a code point to a buffer.
```
Comment (phated) [inline stdlib/buffer.gr:326]:
```suggestion
 * Appends a character represented by a code point a buffer.
```

**Review by phated — CHANGES_REQUESTED** [review]
Doc could use some work

File: stdlib/buffer.gr (hunk 2 above)
Comment (spotandjake) [reply stdlib/buffer.gr]: I made this change, one note is should we refer to the `char` as a `character` in `addchar`? 
currently we say `Appends the bytes of a char to a buffer.`

Comment (phated) [reply stdlib/buffer.gr]: Sure, we use full words for descriptions, but that is unrelated to this PR.

File: stdlib/buffer.gr [hunk 3]
```diff
@@ -322,6 +322,18 @@ provide let addChar = (char, buffer) => {
   addString(Char.toString(char), buffer)
 }
 
+/**
+ * Appends a character represented by a code point a buffer.
```
Comment (phated) [inline stdlib/buffer.gr:326]:
```suggestion
 * Appends a character represented by a code point to a buffer.
```
Looks like we lost this word

**Review by phated — APPROVED** [review]

**Pull request merged**